### DISCUSSION TITLE
Get dimensions of a single cell in pixels?

### DISCUSSION THREAD

**Discussion by colin-heffernan** [wezterm/wezterm#3750] [Q&A] [answered]
I'm trying to make a function to set cell width/height to stretch to fit the window, but for that I need to know the cell's dimensions. Is this possible?

**Comment by wez** [comment]
Can you provide more context?
Are you talking about applications running inside the terminal, or something else?

**Reply by colin-heffernan** [reply]
Sorry for the lack of clarity. I'm talking about the way the terminal itself renders text. Currently, I have my `window_padding` set to completely 0s, which aligns the cells of the terminal to the upper left. This leaves a gap on the bottom and right of the terminal with the leftover space. I am trying to make a function that updates `cell_width` and `line_height` to stretch the cells to fit the entire window. However, to calculate what to set the `cell_width` and `line_height` to, I am trying to determine how large those gaps are. My current approach is to calculate the amount of terminal space taken up by cells. To calculate this, I multiply the height of a single cell by the number of rows. However, I am unable to get the height of a cell. I am trying to find out if there is a way get the dimensions, in pixels, of a single cell.

Sorry if I'm still unclear, I'm not the greatest with explanations. I can try to clarify again, if need be.

**Comment by colin-heffernan** [comment, marked as answer]
Update: I've found that `pane:tab():get_size()` can get the size of the cell section, but the stretch function can't line up exactly with the edges of the window. Thanks!

**Reply by colin-heffernan** [reply]
If you need the size of a single cell, divide `pane:tab():get_size().pixel_width` by `pane:tab():get_size().cols` or `pane:tab():get_size().pixel_height` by `pane:tab():get_size().rows`.
